src/state/ledger/mod.rs

```diff
@@ -327,11 +340,16 @@ mod tests {
             .apply_diff(&ledger_diff)
             .expect("ledger diff application");
 
-        let account_before = account.clone();
         let account_after = ledger.accounts.get(&public_key).expect("account get");
 
-        assert_eq!(account_before.balance, Amount(10)); // with cloning
-        assert_eq!(account_after.balance, Amount(11)); // Assert against the balance
+        if let AccountDiff::Payment(payment_diff) = &ledger_diff.account_diffs[0] {
```

Comment:
Please delete the `if ... else ...`, it is definitely and `AccountDiff`

Reply:
Since payment_diff is not a direct field of AccountDiff, we may need to ```if else or ```match the enum variant to access it.